### DISCUSSION TITLE
request: ページの入力文字制限（正規表現）を行いたい

### DISCUSSION THREAD

**Discussion by michi-take** [growilabs/growi#7714] [Ideas / Requests]
Environment
------------

### Host

| item     | version |
| ---      | --- |
|OS        |RHEL7|
|GROWI     |4.0.0|
|Using Docker|yes|
|Using [growi-docker-compose][growi-docker-compose]|yes|

[growi-docker-compose]: https://github.com/weseek/growi-docker-compose

*(Accessing https://{GROWI_HOST}/admin helps you to fill in above versions)*

### Client

| item     | version |
| ---      | --- |
|OS        |Windows10|
|browser   |chrome|

How to reproduce? (再現手順)
---------------------------

1. ページを作成・編集する。
    ```bash
    
    ```

What is the expected result? (期待される動作)
-------------------------------------------

- ページに入力した文字列が制限される。ようにしてほしい。
　事前に、入力を制限した文字列（正規表現）を設定しておき、
　その文字がページに記載されると、「更新」ボタンを押下した際に、
　エラーメッセージが表示され保存できない。

Note
----
現在、社内のシステム部でのみGrowiを利用していますが、
近い将来（秋頃）に、他部署を含めた、社内全体のWikiツールとして利用を計画しています。
ただ、中にはお客さまの契約情報を扱う部署が存在するため、利用ルールとともに、
Growi自体でも、誤って契約番号などを載せないように抑止できるようにしたいと思っています。ランダムな数字などが存在するため、正規表現などでチェックできる機能があると嬉しいです。
利用者の規模がとても大きくなり、弊社としとても重要な位置付けの機能となるため、ぜひ機能追加を検討して頂きたいです。

**Comment by yuki-takei** [comment]
@michi-take 
ご意見ありがとうございます。

あまり耳にしたことのない要望なので、結構ニッチな機能という印象です。難易度は高くないので、チームで一旦検討し、実装スケジュールが立ちましたら WIP ラベルをつけます。(が、あまり期待しないで待ってていただけると…)

一応管理画面のカスタムスクリプト等で自前実装は可能だとは思いますので、JS を使える方がいらっしゃるようでしたらそちらも検討してみてください。

**Comment by yuki-takei** [comment]
別の道というのも変ですが、GROWI.cloud のビジネスプランをご利用の場合はそちらの優先サポート経由でご要望いただくことで本体機能に関しても実装を優先して進める場合があります。併せてご検討いただければ幸いです。

**Comment by michi-take** [comment]
ご検討ありがとうございます。よろしくお願いいたします。
また、GROWI.cloudのビジネスプランの利用について、チーム内でご検討いたします。
